src/Orleans/Messaging/Message.cs

```diff
@@ -440,24 +444,15 @@ public Message()
             headerBytes = null;
         }
 
-        private Message(Categories type, Directions subtype)
-            : this()
-        {
-            Category = type;
-            Direction = subtype;
-        }
-
         internal static Message CreateMessage(InvokeMethodRequest request, InvokeMethodOptions options)
         {
-            var message = new Message(
```

Comment:
I have that property locally, but I thought that cost of Interlocked increment would be high enough to be noticed. But now i see that it's cost is negligible.